### PR TITLE
[DEVX-1064] Add docs for multi-app support on saucectl

### PR DESCRIPTION
**Pull request by tianfeng92**
**Thank you for contributing to the `sauce-docs` project!**
**A well described pull request helps maintainers quickly review and merge your change**

Before submitting your PR, please check our [contributing](https://github.com/saucelabs/sauce-docs/blob/master/CONTRIBUTING.MD)
guidelines, applied for this repository. Avoid large PRs, help reviewers by making them as simple 
and short as possible.

Add docs changes on saucectl side for supporting multi-apps testing.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation fix (typos, incorrect content, missing content, etc.)

### Checklist

- [ ] I have read the [contributing](https://github.com/saucelabs/sauce-docs/blob/master/CONTRIBUTING.MD) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: docs/mobile-apps/automated-testing/espresso-xcuitest.md [hunk 1]
```diff
@@ -101,19 +102,19 @@ Both Test Object and Sauce Labs utilize CLI commands and YAML configuration file
 | Choose a tablet device only. | `--tabletOnly` | Must use YAML |
 | Ensure the device is connected to a cellular network. | Not supported | Must use YAML |
 | Choose any device where the name matches the regex. | `--deviceNameQuery` | Must use YAML |
-| Specify which test class to run. | `--testsToRun <class>` | Must use YAML |
+| Specify which test class to run. | `--testsToRun <class>` | `--testOptions.class` |
```
Comment (alexplischke) [inline docs/mobile-apps/automated-testing/espresso-xcuitest.md:105]: That only works though if the user is using the CLI driven approach, which we have not yet documented. So you can't specify `saucectl run --testOptions.class`. Same goes for all the other framework specify flags.

Comment (tianfeng92) [reply docs/mobile-apps/automated-testing/espresso-xcuitest.md:105]: ok.. I'll remove these changes

File: docs/testrunner-toolkit/configuration/espresso.md [hunk 2]
```diff
@@ -246,6 +249,23 @@ The path to the testing application. The relative file location is `{project-roo
 ```
 ---
 
+### `otherApps`
+<p><small>| OPTIONAL | ARRAY |</small></p>
+
+List of paths to apps to be pre-installed for running tests. The relative file location is `{project-root}/apps/app1.apk`, and the property supports expaned environment vairables to designate the path, as shown in the following examples.
```
Comment (sweeneyskirt-sl) [inline docs/testrunner-toolkit/configuration/espresso.md:255]:
```suggestion
Set of one or more paths to apps to be pre-installed for running tests. The relative file location is `{project-root}/apps/app1.apk`, and the property supports expanded environment variables to designate the path, as shown in the following examples.
```

File: docs/testrunner-toolkit/configuration/xcuitest.md [hunk 3]
```diff
@@ -246,6 +249,23 @@ The path to the testing application. The property recognizes both `.ipa` and `.a
 ```
 ---
 
+### `otherApps`
+<p><small>| OPTIONAL | ARRAY |</small></p>
+
+List of paths to apps to be pre-installed for running tests. The relative file location is `{project-root}/apps/app1.ipa`, and the property supports expaned environment vairables to designate the path, as shown in the following examples.
```
Comment (sweeneyskirt-sl) [inline docs/testrunner-toolkit/configuration/xcuitest.md:255]:
```suggestion
Set of one or more paths to apps to be pre-installed for running tests. The relative file location is `{project-root}/apps/app1.ipa`, and the property supports expanded environment variables to designate the path, as shown in the following examples.
```

File: docs/testrunner-toolkit/configuration/espresso.md (hunk 2 above)
Comment (tianfeng92) [reply docs/testrunner-toolkit/configuration/espresso.md:255]: ok!

**Review by sweeneyskirt-sl — APPROVED** [review]
:shipit:

**Comment by JudgeSi** [comment]
Hi @tianfeng92 thanks for this! 
Just to make sure, you are now only writing the docs for the multi app support that are concerning saucectl correct? 

I think than it might make sense to split the original ticket up, since we might want to mention this feature in other places also.

File: docs/testrunner-toolkit/configuration/espresso.md (hunk 2 above)
Comment (JudgeSi) [reply docs/testrunner-toolkit/configuration/espresso.md]: Actually the customer is supposed to not only be able to pass in filepaths inside the `otherApps` capability, but also AppStorage URIs directly. (i.e. strings prefixed with `storage:` this is so we have feature parity to appium.

I am sorry if this was not clear from the tickets, it is mentioned though on https://saucedev.atlassian.net/browse/DEVX-1064

**Review by JudgeSi — COMMENTED** [review]
We are missing a section which explains which device/app settings will be applied to the `otherApps`, see also https://saucedev.atlassian.net/browse/DEVX-1064 
This is important documentation for our customers so they are aware of the edge cases in multi app support. 

It might make sense though to not place this section in the saucectl reference. But we should have a discussion where it should be placed. Let me know if we should sync on this.

File: docs/testrunner-toolkit/configuration/espresso.md [hunk 4]
```diff
@@ -246,6 +249,23 @@ The path to the testing application. The relative file location is `{project-roo
 ```
 ---
 
+### `otherApps`
+<p><small>| OPTIONAL | ARRAY |</small></p>
+
+Set of one or more paths to apps to be pre-installed for running tests. The relative file location is `{project-root}/apps/app1.apk`, and the property supports expanded environment variables to designate the path, as shown in the following examples.
```
Comment (JudgeSi) [inline docs/testrunner-toolkit/configuration/espresso.md:255]: Actually the customer is supposed to not only be able to pass in filepaths inside the otherApps capability, but also AppStorage URIs directly. (i.e. strings prefixed with storage: this is so we have feature parity to appium.

I am sorry if this was not clear from the tickets, it is mentioned though on https://saucedev.atlassian.net/browse/DEVX-1064

**Comment by sweeneyskirt-sl** [comment]
> It might make sense though to not place this section in the saucectl reference. But we should have a discussion where it should be placed. Let me know if we should sync on this.

Hi @JudgeSi -- thanks for the feedback. So even though these docs are organized in the original saucectl directory of our repo, we have recently made changes to the navigation in our site such that these pages show up in the [mobile-apps/automated-testing/espresso-xcuitest](https://docs.saucelabs.com/mobile-apps/automated-testing/espresso-xcuitest) section of the docs, which I think is in alignment with your thoughts.

**Comment by sweeneyskirt-sl** [comment]
> We are missing a section which explains which device/app settings will be applied to the `otherApps`, see also https://saucedev.atlassian.net/browse/DEVX-1064
> This is important documentation for our customers so they are aware of the edge cases in multi app support.

@tianfeng92 and I will review this additional information to clarify the property description. Thanks for the heads up, @JudgeSi

**Comment by JudgeSi** [comment]
Hi @sweeneyskirt-sl  @tianfeng92 

I had a look again at the existing documentation for the app settings and this might be a larger task than expected. So I suggest for now that we tackle it in the ticket that @tianfeng92 created: https://saucedev.atlassian.net/browse/DEVX-1100 
I will add all my pointers to this ticket and then we can move it to MOBA, as I think that there are also a few things we might want to discuss with Lina. 

This way we can merge this here as it is now and move on.

**Comment by tianfeng92** [comment]
Hi @JudgeSi !
I agree with your comment and we can merge this for saucectl side.

File: docs/testrunner-toolkit/configuration/espresso.md (hunk 4 above)
Comment (tianfeng92) [reply docs/testrunner-toolkit/configuration/espresso.md:255]: This ticket is for adding storage id https://saucedev.atlassian.net/browse/DEVX-1095 We can merge this and do more changes upon it.

**Pull request merged**